lib/Parse/ParseType.cpp

```diff
@@ -670,19 +670,40 @@ ParserResult<TypeRepr> Parser::parseTypeIdentifier() {
 /// parseTypeSimpleOrComposition
 ///
 ///   type-composition:
-///     type-simple
+///     'some'? type-simple
 ///     type-composition '&' type-simple
 ParserResult<TypeRepr>
 Parser::parseTypeSimpleOrComposition(Diag<> MessageID,
                                      bool HandleCodeCompletion) {
+  // Check for the opaque modifier.
+  // This is only semantically allowed in certain contexts, but we parse it
+  // generally for diagnostics and recovery.
+  SourceLoc opaqueLoc;
+  if (Context.LangOpts.EnableOpaqueResultTypes
+      && Tok.is(tok::identifier)
+      && Tok.getRawText() == "some") {
+    opaqueLoc = consumeToken();
+  }
```

Comment:
Any reason we don't need back tracking scope for this `some`? The following code will have trouble being parsed:
```
class some {}
func foo() -> some { return some() }
```

Reply:
Yes, it should have a backtracking scope.